test/constants/constants_test.rb

```diff
@@ -0,0 +1,41 @@
+# encoding: ASCII-8BIT
+require 'test_helper'
+require 'pwnlib/constants/constants'
+require 'pwnlib/constants/constant'
+
+class ConstantsTest < MiniTest::Test
+  include Pwnlib::Constants::ClassMethod
+  Constant = ::Pwnlib::Constants::Constant
+
+  def test_constant_methods
+    a1 = Constant.new('a', 1)
+    assert_equal(9, Constant.new('c', 9).to_i)
+    assert_equal(3, a1 | 2)
+    assert_operator(Constant.new('ZERO', 0), :==, 0)
+    # test coerce
+    assert_operator(0, :==, Constant.new('ZERO', 0))
+    assert_operator(a1, :==, Constant.new('b', 1))
+    refute_operator(a1, :==, Constant.new('a', 3))
+    assert_equal(3, a1 | Constant.new('a', 2))
+  end
+
+  def test_amd64
+    context.local(arch: 'amd64') do
+      assert_equal('Constant("SYS_read", 0x0)', send(:SYS_read).inspect)
+      assert_equal('__NR_arch_prctl', send(:__NR_arch_prctl).to_s)
```

Comment:
I thought that we'll never use Constant like this?
Maybe just `Constants = Pwnlib::Constants` and test with `Constants.SYS_read` ?